src/Modules/KnowledgeComponents/Tutor.KnowledgeComponents.Core/Domain/KnowledgeMastery/SessionTracker.cs

```diff
@@ -44,19 +54,28 @@ public Result Terminate()
 
     public Result Pause()
     {
-        if (!HasUnfinishedSession) Launch();
+        if (!HasUnfinishedSession) return Result.Fail("No active session to pause.");
         if (IsPaused) return Result.Fail("Session is already paused.");
         
         IsPaused = true;
```

Comment:
State changes should be in the event handling (When method). `IsPaused = true;` is a state change.